### DISCUSSION TITLE
How to create an array of particular sceneitems?

### DISCUSSION THREAD

**Discussion by MarioMey** [upgradeQ/Streaming-Software-Scripting-Reference#20] [Q&A]
I have this code that create a group and put some sceneitems inside. But it does it by creating the group empty and then put every item inside it. I know that I can create the group with items inside at once... but I need to pass an array of sceneitems. I don't know how!

```python

def item_create_group(unused_addr, scene_name, group_name, items=[], visible=True):
	scene, scenes_list = get_scene_context(scene_name)
	if not scene:
		obs.source_list_release(scenes_list)
		print(f'item_create_group() Scene no existe: "{scene_name}"')
		return

	with source_list_auto_release(scenes_list):

		# TO DO
		# scene_items = []
		# new_group = obs.obs_scene_insert_group(scene, group_name, scene_items, len(scene_items))

		new_group = obs.obs_scene_insert_group(scene, group_name, None, 0)
		for item_name in items:
			scene_item = obs.obs_scene_find_source(scene, item_name)
			obs.obs_sceneitem_group_add_item(new_group, scene_item)

		group_id = obs.obs_sceneitem_get_id(new_group)
		return group_id
```

**Comment by MarioMey** [comment]
Discord:

**bfx — Yesterday at 3:57 AM**
@Mario Mey I'm afraid there is no direct solution. The same old problem with pointer-pointer data types in the Python and Lua bindings (I opened a GitHub issue for that, long ago, still trying to determine the best way to answer).
A workaround is to use FFI, check this: https://obsproject.com/forum/threads/how-to-enumerate-script-properties-in-lua.83406/#post-397381

**Reply by upgradeQ** [reply]
This should be possible with [`ctypes`](https://docs.python.org/3/library/ctypes.html)

**Reply by MarioMey** [reply]
Well, thanks... but as groups are very buggy, I won't use them... unless they work correctly.
